src/ert/scheduler/lsf_driver.py

```diff
@@ -227,6 +229,11 @@ async def submit(
         name: str = "dummy",
         runpath: Optional[Path] = None,
     ) -> None:
+        if iens not in self._submit_locks:
+            self._submit_locks[iens] = asyncio.Lock()
+
+        await self._submit_locks[iens].acquire()
```

Comment:
when looking at the code I would prefer to use the:
```python
async with self._submit_locks[iens]:
 ....
```
which then automatically releases the lock

Reply:
I sort of agree, but I dislike very long indented blocks changing so many lines of code..

Reply:
Using async.with now.

Reply:
I think it is required to use the `async with Lock` as this one guarantees to release the lock when exception occurs :+1:

Reply:
fixed.